pulser-core/pulser/waveforms.py

```diff
@@ -295,28 +314,36 @@ def _check_slice(self, s: slice) -> slice:
         return slice(start, stop)
 
     @abstractmethod
-    def __mul__(self, other: float) -> Waveform:
+    def __mul__(self, other: float | ArrayLike) -> Waveform:
         pass
 
     def __neg__(self) -> Waveform:
         return self.__mul__(-1.0)
 
-    def __truediv__(self, other: float) -> Waveform:
-        if other == 0:
+    def __truediv__(self, other: float | ArrayLike) -> Waveform:
```

Comment:
- Isn't AbstractArrayLike better than ArrayLike ?
- Should not we either check for the dimension of the provided AbstractArrayLike, or, if the dimension can be higher than 0, that the length of the other is the same as self, and that it does not contain any 0 value ?

Reply:
- From the discussion above: definitely ArrayLike.
- Ookay, I did not get that __mul__ (and, per extension, __truediv__) was for 1-dimensional elements. It was failing ungracefully before, so I guess let's keep it failing ungracefully. However, even though it's intended to work with scalars only, I think it now works with 1D arrays as well... Do you want to put a condition on the dimension of the arrays in __mul__ ?